Helm/opensearch/templates/statefulset.yaml

```diff
@@ -237,6 +240,43 @@ spec:
           containerPort: {{ .Values.httpPort }}
         - name: transport
           containerPort: {{ .Values.transportPort }}
+        {{- if and .Values.probes.readiness.enabled .Values.probes.readiness.username .Values.probes.readiness.password }}
+        readinessProbe:
+          exec:
+            command:
+            - sh
+            - -c
+            {{- if .Values.probes.readiness.acceptable_statuses }}
+            - 'curl --fail --silent -u $READINESS_USERNAME:$READINESS_PASSWORD http://localhost:9200/_cat/health | grep -P "({{ join "|" .Values.probes.readiness.acceptable_statuses }})"'
+            {{- else }}
+            - 'curl --fail --silent -u $READINESS_USERNAME:$READINESS_PASSWORD http://localhost:9200/_cat/health'
+            {{- end }}
```

Comment:
If we are happy and this always works. We should not let this be conditional. It's helpful and shouldn't really change.

Reply:
@DandyDeveloper if we wanted to leave this always available I suppose we could set the default for the acceptable_statuses to `["red", "yellow", "green"]` though I think having the users opt in to that behavior gives them a greater chance of selecting what makes sense in their environment.

Reply:
@hgoscenski-imanage 

Actually, I'm second guessing myself here. Do we even want this in the chart? Nodes will still accept documents and process when the shards for that index become available. The cluster can still be red but accepting new documents. 

If we implement this, it means even an `Initialized` node, with recovered shards would not be receiving requests because the k8s service will no longer be routing traffic to them (Because ES is reporting things as red). 

If other nodes are available this wouldn't be a problem, but if you do have a complete cluster failure, or you want to rollout a cluster upgrade, larger clusters with this could be waiting days until a node becomes available after recovering all shards. 

As a user, what scenario would you want ["yellow", "green"] to be the only accepted readiness states?

Reply:
@DandyDeveloper I am currently running it in that configuration in our environment, we want to ensure that in combination with the pod disruption budgets that even during automatic draining of nodes that the cluster always remains in an available state. 

This can slow down the rollout process but ensures availability, if the nodes are returning with the `"index.unassigned.node_left.delayed_timeout"` then there would be no unnecessary shuffling of shards.

Our cluster is simply being used to handle logs from a kubernetes cluster and is a relatively small set of data, but I believe there is value in having the ability to configure this. 

All of our log ingesters write to a deployment of opensearch nodes acting as clients which do not check the status of the cluster but rather simply check for the availability of the node. This allow logs to always be written assuming the cluster is maintaining a status where the indices are available, but does imply the necessity of configuration here, at least in our environment.